src/main/java/org/amahi/anywhere/server/client/ServerClient.java

```diff
@@ -210,6 +210,7 @@ public void connectLocal() {
         }
         this.serverAddress = serverRoute.getLocalAddress();
         this.serverApi = buildServerApi();
+        ServerConnectionDetectingTask.execute(serverRoute);
```

Comment:
Why have you added this line here?
This will automatically choose first local address and then switch to remote address if the ping to local address fails. Adding this dissolves the purpose of having `Local` and `Remote` mode explicitly.